lib/url.py

```diff
@@ -66,24 +66,11 @@ def archive_message_url(
 
 ## String cleaning functions
 
-# remove non-alnum ascii symbols from string
 def sanitize(s):
+    # Sanitize the string to a safe string that can be used as the URL
+    # It copies zulip's way of encoding stream and topic name to URL
     return (
-        "".join(
-            filter(
-                lambda x: x.isalnum or x == " ",
-                s.encode("ascii", "ignore").decode("utf-8"),
-            )
-        )
-        .replace(" ", "-")
-        .replace("?", "%3F")
```

Comment:
This needs to be investigated. It's due to #34. GitHub Pages might have a stricter set of URL escaping than Zulip Server.

Reply:
What about the topic sanitization? Does it not include " " -> "-" in the webapp?

Reply:
As far as I see it doesn't

Here's an example, the original topic name is `crab rave`
```
https://chat.zulip.org/#narrow/stream/276-music/topic/crab.20rave
```

Reply:
This is how the web app makes the topic url:

https://github.com/zulip/zulip/blob/de31114d700561f32139a63a0e5f33d5c30039b3/zerver/lib/url_encoding.py#L41